packages/ui/src/Button.tsx

```diff
@@ -92,71 +100,101 @@ export function Button({
   if (color === "gray") {
     color = "lightGray";
   }
+
+  const renderConfirmation = () => {
+    return (
+      <Modal
+        heading={confirmationHeading}
+        primaryButtonOnClick={() => {
+          onClick();
+          setShowConfirmation(false);
+        }}
+        primaryButtonText="Confirm"
+        secondaryButtonOnClick={(): void => setShowConfirmation(false)}
+        secondaryButtonText="Cancel"
+        size="sm"
+        visible={showConfirmation}
+        onDismiss={(): void => {
+          setShowConfirmation(false);
+        }}
+      >
+        <Text>{confirmationText}</Text>
+      </Modal>
+    );
+  };
+
   return (
-    <TouchableOpacity
-      disabled={disabled || loading}
-      style={{
-        alignSelf: inline === true ? undefined : "stretch",
-        height: HEIGHTS[size || "md"],
-        backgroundColor: getBackgroundColor(color),
-        // width: inline === true ? undefined : "100%",
-        flexShrink: inline ? 1 : 0,
-        // flexGrow: inline ? 0 : 1,
-        alignItems: "center",
-        justifyContent: "center",
-        borderRadius: 5,
-        borderColor: getBorderColor(color),
-        borderWidth: type === "outline" ? 2 : 0,
-        opacity: disabled ? 0.4 : 1,
-        flexDirection: "row",
-        paddingHorizontal: 4 * 2,
-      }}
-      onPress={debounce(
-        async () => {
-          Unifier.utils.haptic();
-          setLoading(true);
-          try {
-            if (onClick) {
-              await onClick();
+    <>
+      <TouchableOpacity
+        disabled={disabled || loading}
+        style={{
+          alignSelf: inline === true ? undefined : "stretch",
+          height: HEIGHTS[size || "md"],
+          backgroundColor: getBackgroundColor(color),
+          // width: inline === true ? undefined : "100%",
+          flexShrink: inline ? 1 : 0,
+          // flexGrow: inline ? 0 : 1,
+          alignItems: "center",
+          justifyContent: "center",
+          borderRadius: 5,
+          borderColor: getBorderColor(color),
+          borderWidth: type === "outline" ? 2 : 0,
+          opacity: disabled ? 0.4 : 1,
+          flexDirection: "row",
+          paddingHorizontal: 4 * 2,
+        }}
+        onPress={debounce(
+          async () => {
+            Unifier.utils.haptic();
+            setLoading(true);
+            try {
+              if (withConfirmation) {
+                if (!showConfirmation) {
```

Comment:
You can combine this conditional with the one above: `withConfirmation && !showConfirmation`

Reply:
Ah good catch!!